pkg/controller/workload/kubernetes/application/application.go

```diff
@@ -146,6 +145,21 @@ func (c *localCluster) sync(ctx context.Context, app *v1alpha1.KubernetesApplica
 	return v1alpha1.KubernetesApplicationStateSubmitted, nil
 }
 
+func (c *localCluster) delete(ctx context.Context, app *v1alpha1.KubernetesApplication) (int, error) {
```

Comment:
@muvaf Would it make sense for this to be a method on `applicationResourceSyncer`? That way I believe we'd keep the `client.Client` contained there, rather than needing to add it to the `localCluster` type as well.

Reply:
Actually, I see now that that probably doesn't make sense. Could we instead follow the pattern of `applicationResourceSyncer` with an `applicationResourceDeleter`, i.e. make this a simple injected dependency of `localCluster` rather than a method? I know we don't love the logic that's there now, but it would be ideal to be consistent with the existing patterns and separation of concerns if possible.

Reply:
There are a few abstraction layer here. As far as I see, `applicationResourceSyncer.sync` is called for an individual KAR while `localCluster.sync` is called for the whole KA, hence it's running a `for` loop to iterate through the KARs.

For deletion, we need to call the following per resource:
```
err := c.kube.Delete(ctx, kar)
```

If I understand what you're saying, you'd like `applicationResourceDeleter` on `localCluster` with a `delete` function that only has `c.kube.Delete(ctx, kar)` and then we'd have `localCluster.delete` to call that function like `c.ad.delete` for every KAR, right?

Isn't that kind of too much for a function with one delete call?

Reply:
> Isn't that kind of too much for a function with one delete call?

I agree it's too many layers of abstraction. I have a PR (pending Github 500 issues) that removes a layer of abstraction.